File: scripts/Engineering/gui/engineering_diffraction/settings/settings_view.py
# Mantid Repository : https://github.com/mantidproject/mantid
#
# Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
#   NScD Oak Ridge National Laboratory, European Spallation Source,
#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
# SPDX - License - Identifier: GPL - 3.0 +
from qtpy import QtWidgets, QtCore

from mantidqt.utils.qt import load_ui

Ui_settings, _ = load_ui(__file__, "settings_widget.ui")


class SettingsView(QtWidgets.QDialog, Ui_settings):
    def __init__(self, parent=None):
        super(SettingsView, self).__init__(parent)
        self.setWindowFlags(self.windowFlags() & ~QtCore.Qt.WindowContextHelpButtonHint)
        self.setupUi(self)
        self.setModal(True)

        self.finder_save.setLabelText("Save Location")
        self.finder_save.isForRunFiles(False)
        self.finder_save.isForDirectory(True)

        self.finder_fullCalib.setLabelText("Full Calibration")
        self.finder_fullCalib.isForRunFiles(False)

    # ===============
    # Slot Connectors
    # ===============

    def set_on_apply_clicked(self, slot):
        self.btn_apply.clicked.connect(slot)

    def set_on_ok_clicked(self, slot):
        self.btn_ok.clicked.connect(slot)

    def set_on_cancel_clicked(self, slot):
        self.btn_cancel.clicked.connect(slot)

    def set_on_log_changed(self, slot):
        self.log_list.itemChanged.connect(slot)

    # =================
    # Component Getters
    # =================

    def get_save_location(self):
        return self.finder_save.getFirstFilename()

    def get_full_calibration(self):
        return self.finder_fullCalib.getFirstFilename()

    def get_van_recalc(self):
        return self.check_vanRecalc.isChecked()

    def get_checked_logs(self):
        return ','.join([self.log_list.item(ilog).text() for ilog in range(self.log_list.count()) if
                         self.log_list.item(ilog).checkState() == QtCore.Qt.Checked])

    # =================
    # Component Setters
    # =================

    def set_save_location(self, text):
        self.finder_save.setText(text)

    def set_full_calibration(self, text):
        self.finder_fullCalib.setText(text)

    def set_van_recalc(self, checked):
        self.check_vanRecalc.setChecked(checked)

    def add_log_checkboxs(self, logs):
        for log in logs.split(','):
            item = QtWidgets.QListWidgetItem(self.log_list)
            item.setText(log)
            item.setCheckState(QtCore.Qt.Unchecked)
            self.log_list.addItem(item)

    def set_checked_logs(self, logs):
        for log in logs.split(','):
            items = self.log_list.findItems(log, QtCore.Qt.MatchExactly)
            items[0].setCheckState(QtCore.Qt.Checked)

    # =================
    # Force Actions
    # =================

    def find_full_calibration(self):
        self.finder_fullCalib.findFiles(True)

    def find_save(self):
        self.finder_save.findFiles(True)

File: qt/widgets/spectroscopy/src/SettingsWidget/SettingsView.cpp
// Mantid Repository : https://github.com/mantidproject/mantid
//
// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
//   NScD Oak Ridge National Laboratory, European Spallation Source,
//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
// SPDX - License - Identifier: GPL - 3.0 +
#include "MantidQtWidgets/Spectroscopy/SettingsWidget/SettingsView.h"
#include "MantidQtWidgets/Spectroscopy/SettingsWidget/SettingsPresenter.h"

#include "MantidQtWidgets/Common/HelpWindow.h"

namespace MantidQt::CustomInterfaces {

SettingsView::SettingsView(QWidget *parent)
    : QWidget(parent), m_presenter(), m_uiForm(std::make_unique<Ui::InterfaceSettings>()) {
  m_uiForm->setupUi(this);

  connect(m_uiForm->pbOk, &QPushButton::clicked, this, &SettingsView::notifyOkClicked);
  connect(m_uiForm->pbApply, &QPushButton::clicked, this, &SettingsView::notifyApplyClicked);
  connect(m_uiForm->pbCancel, &QPushButton::clicked, this, &SettingsView::notifyCancelClicked);
  connect(m_uiForm->pbHelp, &QPushButton::clicked, this, &SettingsView::openHelp);
}

QWidget *SettingsView::getView() { return this; }

void SettingsView::subscribePresenter(SettingsPresenter *presenter) { m_presenter = presenter; }

void SettingsView::notifyOkClicked() { m_presenter->notifyOkClicked(); }

void SettingsView::notifyApplyClicked() { m_presenter->notifyApplyClicked(); }

void SettingsView::notifyCancelClicked() { m_presenter->notifyCancelClicked(); }

void SettingsView::openHelp() {
  MantidQt::API::HelpWindow::showCustomInterface(QString("Indirect Settings"), QString("indirect"));
}

void SettingsView::setSelectedFacility(QString const &text) {
  auto const index = m_uiForm->cbFacility->findText(text);
  m_uiForm->cbFacility->setCurrentIndex(index != -1 ? index : 0);
}

QString SettingsView::getSelectedFacility() const { return m_uiForm->cbFacility->currentText(); }

void SettingsView::setRestrictInputByNameChecked(bool check) { m_uiForm->ckRestrictInputDataNames->setChecked(check); }

bool SettingsView::isRestrictInputByNameChecked() const { return m_uiForm->ckRestrictInputDataNames->isChecked(); }

void SettingsView::setPlotErrorBarsChecked(bool check) { m_uiForm->ckPlotErrorBars->setChecked(check); }

bool SettingsView::isPlotErrorBarsChecked() const { return m_uiForm->ckPlotErrorBars->isChecked(); }

void SettingsView::setLoadHistoryChecked(bool check) { m_uiForm->ckLoadHistory->setChecked(check); }

bool SettingsView::isLoadHistoryChecked() const { return m_uiForm->ckLoadHistory->isChecked(); }

void SettingsView::setDeveloperFeatureFlags(QStringList const &flags) {
  m_uiForm->leDeveloperFeatureFlags->setText(flags.join(" "));
}

QStringList SettingsView::developerFeatureFlags() const { return m_uiForm->leDeveloperFeatureFlags->text().split(" "); }

void SettingsView::setApplyText(QString const &text) { m_uiForm->pbApply->setText(text); }

void SettingsView::setApplyEnabled(bool enable) { m_uiForm->pbApply->setEnabled(enable); }

void SettingsView::setOkEnabled(bool enable) { m_uiForm->pbOk->setEnabled(enable); }

void SettingsView::setCancelEnabled(bool enable) { m_uiForm->pbCancel->setEnabled(enable); }

} // namespace MantidQt::CustomInterfaces
